### PR TITLE
[tflite] add xnnpack delegate to label_image

### PR DESCRIPTION
**Pull request by freedomtan**
rebase and resubmit #36749 to see if it works.

The [XNNPACK Delegate](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/lite/delegates/xnnpack) uses [XNNPACK](https://github.com/google/XNNPACK), a fairly optimized floating point library, to run some inference operators (see the delegate's [README](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/delegates/xnnpack/README.md) for currently supported ops). With XNNPACK, I was able to get performance numbers similar to what @Maratyszcza described at [XNNPACK's readme](https://github.com/google/XNNPACK/README.md). I also got good numbers on Pixel 4 and Oppo Reno 3. Numbers on x86 machines are also good.

- **Single-threaded**, `label_image -m MODEL_NAME -x 1 -c 50 -t 1`

Model |  Pixel 3a, ms | Pixel 4, ms| Oppo Reno 3, ms |
-- | --: | --: | --:
MobileNet v1 1.0X |  88.02 | 29.95 | 36.23 |
MobileNet v2 1.0X |  55.13 | 19.13 | 21.69 |
MobileNet v3 Large |  44.84 | 15.69 | 17.65 |
MobileNet v3 Small |   14.23 | 5.58 | 5.66|

- **multi-threaded**, the number of threads is set to be the number of big cores. `label_image -m MODEL_NAME -x 1 -c 50 -t NUMBER_OF_BIG_CORES`

Model |  Pixel 3a, ms | Pixel 4, ms| Oppo Reno 3, ms|
-- | --: | --: | --:
MobileNet v1 1.0X |  46.00 | 10.60 | 13.23 |
MobileNet v2 1.0X |  28.57 | 6.89 | 7.56 |
MobileNet v3 Large |  24.62 | 6.52 | 6.88 |
MobileNet v3 Small |   8.24 | 2.47 |  2.37 |

### CODE REVIEW

File: tensorflow/lite/examples/label_image/label_image.cc [hunk 1]
```diff
@@ -101,6 +102,19 @@ TfLiteDelegatePtrMap GetDelegates(Settings* s) {
     }
   }
 
+  if (s->xnnpack_delegate) {
+    TfLiteXNNPackDelegateOptions xnnpack_options =
+        TfLiteXNNPackDelegateOptionsDefault();
+    xnnpack_options.num_threads = s->number_of_threads;
+
+    auto delegate = evaluation::CreateXNNPACKDelegate(&xnnpack_options);
+    if (!delegate) {
```
Comment (multiverse-tf) [inline tensorflow/lite/examples/label_image/label_image.cc]: Might simply call "evaluation::CreateXNNPACKDelegate(s->number_of_threads)"?

File: third_party/cpuinfo/BUILD.bazel [hunk 2]
```diff
@@ -158,6 +160,18 @@ cc_library(
 
 ############################# Build configurations #############################
 
+config_setting(
+    name = "linux_aarch64",
```
Comment (multiverse-tf) [inline third_party/cpuinfo/BUILD.bazel]: This part has already been added in https://github.com/tensorflow/tensorflow/commit/4f395601839390b08da3323a0608a9f22d3d0ac3

File: tensorflow/lite/examples/label_image/label_image.cc (hunk 1 above)
Comment (freedomtan) [reply tensorflow/lite/examples/label_image/label_image.cc:107]: thanks. done.

File: third_party/cpuinfo/BUILD.bazel (hunk 2 above)
Comment (freedomtan) [reply third_party/cpuinfo/BUILD.bazel]: removed.

**Review by multiverse-tf — APPROVED** [review]

**Pull request merged**